etl/harmonize.py

```diff
@@ -70,20 +75,45 @@ def harmonize(
 
     If a mapping file already exists, it will resume where the mapping file left off.
     """
-
+    # Load
     df = read_table(data_file)
-    geo_column = cast(pd.Series, df[column].dropna().astype("str"))
 
-    if Path(output_file).exists():
-        print("Resuming from existing mapping...\n")
-        with open(output_file, "r") as istream:
-            mapping = json.load(istream)
-    else:
-        mapping = {}
+    # Create Harmonizer
+    harmonizer = Harmonizer(
+        tb=df,
+        colname=column,
+        output_file=output_file,
+    )
+
+    # Run automatic harmonization
+    ambiguous = harmonizer.run_automatic(logging="shell")
+
+    # Need user input
+    harmonizer.run_interactive_terminal(ambiguous, institution, num_suggestions)
+
+    # Export
+    harmonizer.export_mapping()
 
-    mapping = interactive_harmonize(geo_column, mapping, institution=institution, num_suggestions=num_suggestions)
-    with open(output_file, "w") as ostream:
-        json.dump(mapping, ostream, indent=2)
+
+def harmonize_ipython(
+    tb: Table, column: str, output_file: str, num_suggestions: int = 100, institution: Optional[str] = None
```

Comment:
Argument "column" should not be mandatory. Its default value can be defined as a global variable. Alternatively, you can make it None by default (an also None by default in Harmonizer, and if None, then its value is set to "country").

Reply:
Ideally, argument `output_file` should also be optional. At some point in the future, the path to the excluded countries should also become an argument. Maybe `paths` could be given as an optional argument, instead. If that's too much hassle, you can leave it this way.

Reply:
you need to provide a value to the export file. I don't see how you can avoid that.

It's either as-is now, or provide paths and access the country mapping file from within the class.

I think the current state is more general and flexible, since it can be used regardless of whether you have paths or not.

What I can do, is adding an additional argument to accept paths alternatively

Reply:
> Argument "column" should not be mandatory. Its default value can be defined as a global variable. Alternatively, you can make it None by default (an also None by default in Harmonizer, and if None, then its value is set to "country").

do you mean to basically have the default 'country' value?

I think that's a good option.

Reply:
Yes, that's what I mean. I thought that maybe you didn't do it that way because the Harmonizer already has "country" defined as a default argument. For that reason I suggested defining a global variable or setting it to None. But it's definitely not necessary (in most cases, indeed, the column name is going to be "country" anyway).